### PR TITLE
Scroll to validation errors and fade error alerts in

### PR DESCRIPTION
**Pull request by stefan-burke**
## What this changes

When a form submission fails, the page re-renders with the error message shown — but the browser leaves you scrolled at the top of the page. If the error is further down the form (for example the "A start date is required for the booked daily listings" message on the attendee booking form), it's easy to miss it and wonder why nothing happened.

Two small changes make errors clearer:

- **Anchor to the error you just triggered — with no JavaScript.** Each freshly-raised error alert on the attendee form is now focusable (`tabindex="-1"` + `autofocus`). The browser focuses the first one on load and scrolls straight to it. Because only the first `autofocus` element on the page takes focus and the alerts render top-to-bottom, this lands on the topmost error. The name field gives up its own default autofocus whenever the form has an error, so it doesn't win that race over an error lower down. A clean form still focuses the name field as before.
- **Fade errors in with a glow.** Error messages fade in with a brief glow that fades back out, so they stand out from the rest of the form. People who've asked their device to reduce motion see no animation. (Pure CSS.)

Standing notes that happen to be styled like errors (e.g. the money-ledger caution above the date field) deliberately stay plain — no `autofocus` — so they're never a focus target.

## How it works

- `errorAlertAttrs` in `attendee-form.tsx` collects the shared alert attributes (`class="error" role="alert" tabindex="-1" autofocus`); the five fresh-error sites (attendee, date, form-wide, save, per-line) spread it. `formHasError` decides whether the name field keeps its autofocus.
- The fade-and-glow is a CSS keyframe on error alerts in `style.scss`, wrapped in a reduced-motion guard.

No client-side script is involved in the scroll — an earlier JavaScript version was replaced, and its module, test, and bundle wiring were removed.

## Tests

- `server-attendee-form.test.ts` asserts that a failed daily-booking submit renders the date error as a focusable alert and drops the name field's autofocus, and that a clean create form keeps the name field's autofocus with no error claiming it.
- Verified in a real Chromium browser: on an error render the browser focuses the error and scrolls it into view; on a clean form the name field keeps focus.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/ui/client/admin/scroll-to-error.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export const initScrollToError = (): void => {
+  // Submit events bubble to the document; no handler in the app stops them, so
+  // the default (bubble) phase reliably catches every form send.
+  document.addEventListener("submit", rememberSubmit);
+  if (takeSubmitFlag() === null) return;
+  // A success/info flash means the submit worked — don't chase an error.
+  if (document.querySelector('.success[role="alert"], .info[role="alert"]')) {
+    return;
+  }
+  const alert = document.querySelector<HTMLElement>('.error[role="alert"]');
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Good catch — confirmed: in `AttendeeEditForm`, `StatusAndBalanceFields` (with the standing balance-ledger note) renders at line 612, before `SharedDateFields` (the `dateError`) at line 676, so the old "first `.error[role="alert"]`" logic landed on the standing note.

Fixed in 59c2809 without touching the template: the submit listener now snapshots the error alerts already on the page just before the form is sent, and the re-render scrolls to the first alert that was **not** in that snapshot — i.e. the freshly-raised one. The standing note is in the snapshot, so it's skipped, and the date error (new) is chosen. Covered by the "scrolls to the freshly-raised error, skipping the standing note" test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    return flag;
+  } catch {
+    return null;
+  }
+};
+
+export const initScrollToError = (): void => {
+  // Submit events bubble to the document; no handler in the app stops them, so
+  // the default (bubble) phase reliably catches every form send.
+  document.addEventListener("submit", rememberSubmit);
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Fixed in 59c2809. `rememberErrorsOnSubmit` now returns early when `event.defaultPrevented` is true, so a form that cancels its own submit to post via fetch (like the scanner check-in) no longer leaves a stale snapshot behind. Covered by the "ignores a client-cancelled submit" test, which submits with `preventDefault()` and asserts the later error is not chased.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+export const initScrollToError = (): void => {
+  // Submit events bubble to the document; no handler in the app stops them, so
+  // the default (bubble) phase reliably catches every form send.
+  document.addEventListener("submit", rememberErrorsOnSubmit);
+  const before = takeSnapshot();
+  if (before === null) return; // this load did not follow a submit
+  const wasThereBefore = new Set(before);
+  const fresh = errorAlerts().find(
+    (alert) => !wasThereBefore.has(signatureOf(alert)),
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Right — the snapshot-on-submit approach conflated "standing note" with "validation error that persisted across a retry". Fixed in 29fc784 by switching to a baseline captured from the *plain (non-submit) page load* rather than from each submit:

- On a plain load, the alerts showing are the view's standing notes → recorded as the baseline.
- On a submit re-render, scroll to the first alert **not** in that baseline. Crucially, the baseline is left untouched on submit re-renders, so it stays as the original standing-notes set across the whole retry sequence. In your scenario, after fixing the date and resubmitting, the surviving quantity error is still not in the baseline, so it's found and scrolled to.

Covered by the new "scrolls to an error that persists across a retry" test (two errors → fix the first → asserts the retry scrolls to the second).

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Good point — the baseline captured on the recalculate page wouldn't contain the edit page's `MoneyAdjustSection` warning, so after the redirect that warning would look "fresh". Fixed in 29fc784 by restoring the success/info-flash guard: before selecting a fresh error, if the page shows a `.success[role="alert"]` or `.info[role="alert"]` flash, the submit succeeded and we don't scroll. The modifier recalculate redirect renders a `<Flash success>`, so it's caught. Covered by the "does not scroll when a success flash is present (redirect target)" test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  top: number,
+  bottom: number,
+  viewportHeight: number,
+): boolean => top < 0 || bottom > viewportHeight;
+
+/** On a real (not client-cancelled) submit, record that a submit happened so
+ * the next load knows to look for a freshly-raised error. */
+const rememberSubmit = (event: Event): void => {
+  if (event.defaultPrevented) return;
+  safeWrite((store) => store.setItem(SUBMITTED_KEY, "1"));
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Fixed in 676e41e. `rememberSubmit` now only records the marker when the submitting form's method is POST, so GET filter/navigation forms (like the availability checker's GET to `/admin/attendees/new`) don't send the destination through the submit path — that page still records its own standing notes as the baseline. Covered by the new "ignores a GET form submit" test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  document.addEventListener("submit", rememberSubmit);
+
+  const justSubmitted = safeRead(SUBMITTED_KEY) !== null;
+  safeWrite((store) => store.removeItem(SUBMITTED_KEY));
+
+  if (!justSubmitted) {
+    // Plain load: the alerts showing now are the view's standing notes. Record
+    // them as the baseline the next submit re-render diffs against; don't scroll.
+    const notes = JSON.stringify(errorAlerts().map(signatureOf));
+    safeWrite((store) => store.setItem(BASELINE_KEY, notes));
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Fixed in 676e41e. `initScrollToError` now also registers a `pageshow` listener; on a persisted `pageshow` (a bfcache restore, where the script itself doesn't re-execute) it re-records the baseline from the restored page. So in your scenario — attendee edit → another admin page → Back → resubmit — the baseline is refreshed to the attendee page's standing notes before the submit, and the off-screen validation error is correctly identified as fresh. Covered by the new "re-records the baseline on a bfcache restore (persisted pageshow)" test (and a companion asserting a non-persisted pageshow leaves the baseline alone).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+};
+
+export const initScrollToError = (): void => {
+  // Submit events bubble to the document; no handler in the app stops them, so
+  // the default (bubble) phase reliably catches every form send.
+  document.addEventListener("submit", rememberSubmit);
+  // A bfcache restore doesn't re-run this script, so refresh the baseline from
+  // the restored page before the operator submits from it.
+  window.addEventListener("pageshow", (event) => {
+    if ((event as PageTransitionEvent).persisted) recordBaseline();
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Fixed in e9ac23f. Each load now marks the document with a `data-scroll-submit-rerender` attribute on `<html>` when it's a submit re-render. That attribute rides along in the frozen DOM across a bfcache round-trip, so the persisted-`pageshow` handler re-records the baseline only for a plain page — never a failed-submit re-render, whose validation errors are therefore not folded into the standing-note set. In your scenario the baseline stays `{standing note}` across the bfcache restore, so after fixing the date the surviving quantity error is still found and scrolled to. Covered by the new "a bfcache restore of a failed submit keeps the retry working" test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/client/admin/scroll-to-error.ts [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  document.addEventListener("submit", rememberSubmit);
+  // A bfcache restore doesn't re-run this script, so refresh the baseline from
+  // the restored page before the operator submits from it — but only if it was
+  // a plain page, not a failed-submit re-render (whose validation errors must
+  // not be folded into the standing-note baseline, or a retry would miss them).
+  window.addEventListener("pageshow", (event) => {
+    const wasSubmitReRender =
+      document.documentElement.hasAttribute(SUBMIT_RERENDER_ATTR);
+    if ((event as PageTransitionEvent).persisted && !wasSubmitReRender) {
+      recordBaseline();
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/client/admin/scroll-to-error.ts]: Fixed in c349f0d. The `data-scroll-submit-rerender` mark is now set only on a genuine *failed* submit re-render (a submit load with no success/info flash). A successful-submit page carries only standing notes, so it stays unmarked and a bfcache restore re-baselines from it — exactly the distinction you drew. So in the attendee edit flow, the success-redirect edit tab refreshes the baseline on Back, and a subsequent missing-date resubmit correctly scrolls to the date error. Covered by the new "re-baselines a successful-submit page restored from bfcache" test, alongside the existing failed-submit test which confirms failed pages still preserve their baseline.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by stefan-burke** [comment]
Heads up on a direction change (commit e697568): the client-side scroll-to-error module has been **replaced with a no-JavaScript approach** — each fresh error alert on the attendee form is now focusable (`tabindex="-1"` + `autofocus`), so the browser scrolls to the first error natively. This deletes `scroll-to-error.ts`, its test, and the bundle wiring, and removes the whole class of edge cases the earlier Codex threads flagged (those threads are on the now-deleted file, so they're moot). The CSS fade-in animation is unchanged.

Verified end-to-end in Chromium: on an error render the browser focuses the error output and scrolls it into view (name field yields its autofocus); a clean form keeps the name field focused.

Full test suite is green (the only failures locally are the two `stripe-mock.test.ts` binary-*download* tests, which fail only because the sandbox builds the mock from source instead of downloading it — they pass in CI). typecheck, lint, and jscpd all pass.

One caveat, per the maintainer's go-ahead to skip it: this edits `attendee-form.tsx`, and the `precommit:mutation` gate will likely report false survivors on that file — its coverage is spread across several test files, only some of which are in this branch's changed set, so the changed-set mutation check can't see all of them (a documented blind spot). The change itself only adds a shared attribute object, a small pure helper, and attribute spreads; behaviour is covered by the tests above.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/templates/admin/built-sites/panels.tsx [hunk 8]
```diff
@@ -171,9 +171,9 @@ export const SecretsPanel = ({
   if (!view.ok) {
     return (
       <div class="prose">
-        <ErrorAlert
-          message={t("built_sites.secrets_error", { error: view.error })}
-        />
+        <ErrorAlert>
+          {t("built_sites.secrets_error", { error: view.error })}
+        </ErrorAlert>
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/ui/templates/admin/built-sites/panels.tsx]: Good catch — you're right, I misclassified it. `loadSiteSecretsStatus` returning `{ ok: false }` on a plain GET (no hosting ID / unset host key) is a page-state warning, not a submission result, so autofocusing it would scroll past the form on open. Fixed in a8aa5f8 by switching it to `ErrorNote` (calm, no autofocus/role="alert"). The genuinely live errors on that flow — flash messages after a save — keep `ErrorAlert`.

---
_Generated by [Claude Code](https://claude.ai/code)_
